### ISSUE TITLE
Find a better way: Create multiple invoices for each outgoing order

### ISSUE THREAD

**Issue by moshababo** [ExchangeUnion/xud#164]
For each outgoing order, create a separate invoice for each peer. They should grouped, to allow batch cancellation, in case that the order will get cancelled (or filled).

On Raiden swaps, attach an invoice to the order as well. It will contain a unique token of the peer's `raidenAddress` + the order id.

**Comment by kilrau** [comment]
We discussed the method you describe a while ago to be able to a) let the maker be the initiator of the swap and b) clearly differentiate incoming swaps and I'd say let's do it like this for pre-alpha and alpha. 

But since it's an essential part of the protocol and potentially hard to change later: rethinking it now I got the following concern: Scalability. Let's say I have 1000 peers. How long will the creation of 1000 invoices take? Let's consider 2 things: future low-end devices and future supernodes which forward orders which means I don't

I thought about it a bit now and couldn't come up with a clean way for the taker to initiate the order BUT the maker being in control of the preimage. Let's revisit this in a later stage.
